src/libraries/System.Security.Cryptography.X509Certificates/tests/DynamicChainTests.cs

```diff
@@ -481,8 +535,16 @@ public static void NameConstraintViolation_ExcludedTree_Dns()
             SubjectAlternativeNameBuilder builder = new SubjectAlternativeNameBuilder();
             builder.AddDnsName("www.example.com");
 
-            // excluded DNS name constraint for example.com.
+            // excluded DNS name constraint for .example.com.
             string nameConstraints = "3012A110300E820C2E6578616D706C652E636F6D";
+            if (PlatformDetection.IsAndroid)
+            {
+                // Android does not consider the constraint as being violated when it has
+                // the leading period. It checks expects the period as part of the left-side
+                // labels and not the constraint when doing validation.
+                // Use an excluded DNS name constraint without the period: example.com
+                nameConstraints = "3012A110300E820C6578616D706C652E636F6D";
```

Comment:
The lengths in this value are invalid (they exceed the payload length).  Here's the corrected version:

```suggestion
                nameConstraints = "3011A10F300D820B6578616D706C652E636F6D";
```